stdlib/public/runtime/StaticBinaryELF.cpp

```diff
@@ -61,6 +64,33 @@ typedef Elf32_Section Elf_Section;
 
 extern const Elf_Ehdr elfHeader asm("__ehdr_start");
 
+#if !HAVE_GETAUXVAL
+static unsigned long getauxval(unsigned long type) {
+  struct AuxvEntry {
+    unsigned long tag;
+    unsigned long value;
+  };
+
+  int fd = open("/proc/self/auxv", O_RDONLY);
+  if (fd < 0) {
+    return 0;
+  }
+
+  AuxvEntry entry;
+  while (read(fd, &entry, sizeof(AuxvEntry)) == sizeof(AuxvEntry)) {
```

Comment:
Is EINTR a threat here?

Reply:
I don't think so. `/proc` VFS is not a "slow" device.

`man 7 signal` says:
> ... the call will fail with the error EINTR:
           * read(2), readv(2), write(2), writev(2), and ioctl(2) calls on "slow" devices.  A "slow" device is one where the I/O call may block for an indefinite time, for example, a terminal, pipe, or socket.   (A disk  is not a slow device according to this definition.)  ...